subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoReportBase.java

```diff
@@ -48,6 +51,7 @@
 @Incubating
 public abstract class JacocoReportBase extends JacocoBase {
 
+    private Set<JacocoTaskExtension> taskExtensions = new HashSet<JacocoTaskExtension>();
```

Comment:
Thanks for the link to the documentation. Given these facts I think it would be best not to merge the proposed change. Users might rightfully get confused about it. I think anyone that really wants to have this behavior can still write a plugin on top of it that enhances the original behavior.

BTW: Can you describe for what purpose you are using the method `getAllClassDirs()`?